docs/framework/react/api/router/useBlockerHook.md

```diff
@@ -7,19 +7,57 @@ The `useBlocker` method is a hook that [blocks navigation](../../guide/navigatio
 
 ## useBlocker options
 
-The `useBlocker` hook accepts a single _optional_ argument, an option object:
+The `useBlocker` hook accepts a single \_required argument, an option object:
 
-### `options.blockerFn` option
+### `options.shouldBlockFn` option
 
-- Optional
+- Required
```

Comment:
if this required now, that's an API break. please either update the docs or make it backwards compatible

Reply:
It should not be an API break since we still allow the old shape of input object
```
export type UseBlockerOpts = {
  shouldBlockFn: BlockerFn

  enableBeforeUnload?: boolean | (() => boolean)
  disabled?: boolean
  withResolver?: boolean
} & { .... }

type LegacyBlockerFn = () => Promise<any> | any
type LegacyBlockerOpts = {
  blockerFn?: LegacyBlockerFn
  condition?: boolean | any
}

 opts?: UseBlockerOpts | LegacyBlockerOpts | LegacyBlockerFn,
```
But when not using the deprecated signature, the function is required. This should be fine no?

Reply:
I think I had to rename the function property to maintain backward compatibility here, but if there is a way to do it I missed, I can fix it.